test/Garnet.test/RespModuleTests.cs

```diff
@@ -0,0 +1,177 @@
+            TestUtils.DeleteDirectory(TestUtils.MethodTestDir);
+            TestUtils.DeleteDirectory(Directory.GetParent(testModuleDir)?.FullName);
+        }
+
+        private string CreateTestModule(string onLoadBody)
+        {
+            var runtimePath = RuntimeEnvironment.GetRuntimeDirectory();
+            var binPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            Assert.IsNotNull(binPath);
+
+            var referenceFiles = new[]
+            {
+                Path.Combine(runtimePath, "System.Collections.dll"),
+                Path.Combine(runtimePath, "System.Private.CoreLib.dll"),
+                Path.Combine(runtimePath, "System.Runtime.dll"),
+                Path.Combine(binPath, "Garnet.server.dll"),
+                Path.Combine(binPath, "Garnet.common.dll"),
+                Path.Combine(binPath, "Tsavorite.core.dll"),
+            };
+
+            var dir1 = Path.Combine(testModuleDir, Path.GetFileName(TestUtils.MethodTestDir));
+
+            Directory.CreateDirectory(testModuleDir);
+            Directory.CreateDirectory(dir1);
+
+            var testFilePath = Path.Combine(TestUtils.MethodTestDir, "TestModule.cs");
+            using (var testFile = File.CreateText(testFilePath))
+            {
+                testFile.WriteLine(
+                    "using Garnet.server.Module;" +
+                    "using System.Collections.Generic;" +
+                    "using Tsavorite.core;" +
+                    "using Garnet;" +
+                    "using Garnet.server;" +
+                    "namespace TestGarnetModule " +
+                    "{ " +
+                    "   public class TestModule : IModule " +
+                    "   { " +
+                    "       public void OnLoad(ModuleLoadContext context, List<string> moduleArgs)" +
+                    "       { " +
+                    onLoadBody +
+                    "       } " +
+                    "   } " +
+                    "}");
+            }
+
+            var modulePath = Path.Combine(dir1, "TestModule.dll");
+            var filesToCompile = new[] {
+                testFilePath,
+                Path.GetFullPath(@"../main/GarnetServer/Extensions/SetIfPM.cs", TestUtils.RootTestsProjectPath),
+                Path.GetFullPath(@"../main/GarnetServer/Extensions/ReadWriteTxn.cs", TestUtils.RootTestsProjectPath)};
+            TestUtils.CreateTestLibrary(null, referenceFiles, filesToCompile, modulePath);
+            return modulePath;
+        }
+
+        [Test]
+        public void TestModuleLoad()
+        {
+            var onLoad =
+                    "            context.Initialize(\"TestModule\", 1); " +
+                    "            context.RegisterCommand(\"TestModule.SetIfPM\", 2, CommandType.ReadModifyWrite, new SetIfPMCustomCommand()," +
+                    "            new RespCommandsInfo{ Name = \"TestModule.SETIFPM\", Arity = 4, FirstKey = 1, LastKey = 1, Step = 1," +
```

Comment:
Why is Arity set to 4, and the second argument earlier is set to 2?

Reply:
Arity includes command name and key as well. That's why the difference in count.